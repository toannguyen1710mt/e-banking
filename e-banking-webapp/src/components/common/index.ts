export * from './Button';
export * from './Text';
export * from './Input';
export * from './RadioButton';
export * from './Navbar';
export * from './MenuDropdown';
export * from './Checkbox';
export * from './Select';
export * from './Calendar';
export * from './WizardForm';
export * from './Chip';
export * from './Table';
export * from './Tabs';
export * from './Modal';
export * from './Toast';
export * from './LoadingIndicator';
export * from './Pagination';
export * from './SubMenu';
