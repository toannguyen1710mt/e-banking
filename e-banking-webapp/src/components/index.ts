export * from './common';
export * from './icons';
export * from './Error';
export * from './NotFound';
export * from './Slideshow';
export * from './AuthContentWrapper';
export * from './LoginForm';
export * from './TargetTile';
export * from './QuickAction';
export * from './SignUpForm';
export * from './Charts';
export * from './Sidebar';
export * from './CreditCard';
export * from './Transfers';
export * from './ExpenseAnalysis';
export * from './StepProgress';
export * from './MyCards';
export * from './BalanceModal';
export * from './MyTarget';
export * from './SettingContainer/Account/ConnectedAccountsTab';
export * from './SettingContainer/General';
export * from './SettingContainer/Account/DeleteAccountTab';
export * from './ServiceCard';
export * from './SettingContainer/Account';
export * from './DueTile';
export * from './MasterCard';
export * from './CardOverview';
export * from './MetricsCard';
export * from './Skeletons';
export * from './TasksWithCalendar';
export * from './ContainerAnalytics';
export * from './AddCreditCardModal';
export * from './InformationCard';
export * from './AddCreditCardModal/AddCreditCard';
export * from './AnalyticsCard';
export * from './DailyLimit';
export * from './TransactionTable';
export * from './TransactionHistory';
export * from './ContainerTransactions';
export * from './MonthYearPicker';
