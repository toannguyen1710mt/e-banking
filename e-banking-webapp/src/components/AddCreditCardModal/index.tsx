'use client';

import { useEffect, useState, useTransition } from 'react';
import Image from 'next/image';
import { z } from 'zod';
import { zodResolver } from '@hookform/resolvers/zod';
import { useForm } from 'react-hook-form';
import { AuthError, Session } from 'next-auth';

// Constants
import { ERROR_MESSAGES, IMAGES } from '@/constants';

// Interfaces
import { IAccount } from '@/interfaces';

// Actions
import { addNewCardByAccountId } from '@/actions';

// Schemas
import { CreditCardSchema } from '@/schemas';

// Services
import { getAccountsByUserId } from '@/services';

// Components
import * as WizardForm from '@/components/common/WizardForm';
import { Modal } from '../common';
import { AddCreditCard } from './AddCreditCard';
import { ConfirmAddCard } from './ConfirmAddCard';
import { toastStore } from '@/utils';

interface IAddCreditCardModalProps {
  session: Session;
  isOpen: boolean;
  onClose: () => void;
}

type FormValues = z.infer<typeof CreditCardSchema>;

export const AddCreditCardModal = ({
  isOpen,
  session,
  onClose,
}: IAddCreditCardModalProps) => {
  const [accounts, setAccounts] = useState<IAccount[]>([]);
  const [isPending, startTransition] = useTransition();

  const form = useForm<FormValues>({
    resolver: zodResolver(CreditCardSchema),
    defaultValues: {
      cardInfo: {
        holderName: '',
        cardNumber: '',
        expireAt: '',
        ccv: '',
      },
      walletType: 'Main',
    },
    reValidateMode: 'onBlur',
    mode: 'onBlur',
  });

  useEffect(() => {
    const fetchData = async () => {
      try {
        const result = await getAccountsByUserId(session.user.id);

        setAccounts(result || []);
      } catch (error) {
        if (error instanceof AuthError) {
          throw ERROR_MESSAGES.GET_ERROR;
        }
      }
    };

    fetchData();
  }, [session.user.id]);

  const onSubmit = async (data: FormValues) => {
    const { holderName, cardNumber, expireAt, ccv } = data.cardInfo;

    const selectedAccount = accounts.find(
      (account) => account.type === data?.walletType,
    );

    try {
      await addNewCardByAccountId(selectedAccount?.documentId as string, {
        holderName,
        cardNumber,
        expireAt,
        ccv,
      });

      startTransition(() => {
        toastStore.showToast(
          ERROR_MESSAGES.ADD_CARD_SUCCESS,
          'success',
          'top-center',
        );

        return onClose();
      });
    } catch (error) {
      toastStore.showToast(
        ERROR_MESSAGES.ADD_CARD_FAILED,
        'error',
        'top-center',
      );

      if (error instanceof AuthError) {
        throw ERROR_MESSAGES.ADD_CARD_FAILED;
      }
    }
  };

  return (
    <Modal
      isOpen={isOpen}
      onClose={onClose}
      size='4xl'
      placement='center'
      classNames={{
        base: 'p-0 rounded-xl bg-primary-200',
      }}
    >
      <div className='flex items-center justify-between'>
        <Image
          className='max-w-[50%] flex-1 px-[31px] py-[87px]'
          src={IMAGES.ADD_CREDIT_CARD_IMAGE}
          width={333}
          height={280}
          alt='Add Credit Card'
        />

        <div className='min-h-[500px] flex-1 rounded-xl bg-white px-4 py-8'>
          <WizardForm.Root
            schema={CreditCardSchema}
            form={form}
            className='flex grow flex-col'
            onSubmit={onSubmit}
          >
            <WizardForm.Step name='cardInfo' key='cardInfo'>
              <AddCreditCard />
            </WizardForm.Step>

            <WizardForm.Step
              name='confirmationDetails'
              key='confirmationDetails'
            >
              <ConfirmAddCard accounts={accounts} isPending={isPending} />
            </WizardForm.Step>
          </WizardForm.Root>
        </div>
      </div>
    </Modal>
  );
};
