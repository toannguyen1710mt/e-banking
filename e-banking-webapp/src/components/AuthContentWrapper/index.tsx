'use client';

import { ReactNode } from 'react';
import { usePathname } from 'next/navigation';
import Link from 'next/link';
import Image from 'next/image';

// Constants
import { IMAGES, ROUTES } from '@/constants';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text } from '../common';

interface AuthContentWrapperProps {
  textFooter?: string;
  textHeading: ReactNode;
  formContent: ReactNode;
  subheading?: string;
}

export const AuthContentWrapper = ({
  textHeading,
  formContent,
  textFooter,
  subheading,
}: AuthContentWrapperProps) => {
  const pathname = usePathname();

  const isOnSignIn = pathname === ROUTES.SIGN_IN;

  const linkDetails = {
    text: isOnSignIn ? 'Register' : 'Sign In',
    href: isOnSignIn ? ROUTES.SIGN_UP : ROUTES.SIGN_IN,
  };

  const classLogo = isOnSignIn
    ? 'mt-0 md:mt-[108px] pb-[56px] md:pb-9'
    : 'pb-[14px] lg:pb-9';
  const classTextHeading = isOnSignIn ? 'mb-7' : 'mb-0 md:mb-7';

  return (
    <div className='mx-auto flex flex-1 flex-col'>
      {/* Logo */}
      <Image
        className={`${classLogo}`}
        src={IMAGES.LOGO}
        width={104}
        height={36}
        alt='Logo EBanking'
      />

      <div className='max-w-[406px] text-center md:text-left'>
        <Text
          as='h1'
          size={TEXT_SIZE['4XL']}
          variant={TEXT_VARIANT.DEFAULT}
          className={`${classTextHeading}`}
        >
          {textHeading}
        </Text>
        <Text
          size={TEXT_SIZE.BASE}
          variant={TEXT_VARIANT.INFO}
          className='mb-4 font-normal leading-[22px]'
        >
          {subheading}
        </Text>
      </div>

      {/* Form */}
      <div className='mx-auto mb-[17px] max-w-[325px] md:mx-0'>
        {formContent}
      </div>

      {textFooter && (
        <Text
          size={TEXT_SIZE.SM}
          variant={TEXT_VARIANT.DEFAULT}
          className='text-center leading-[20px] md:text-left'
        >
          {textFooter}&nbsp;
          <Link
            href={linkDetails.href}
            className='leading-[20px] text-primary-200'
          >
            {linkDetails.text}
          </Link>
        </Text>
      )}
    </div>
  );
};
