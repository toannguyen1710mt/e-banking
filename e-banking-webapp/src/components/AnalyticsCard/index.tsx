'use client';

// Libs
import { Card, CardBody, CardFooter, CardHeader } from '@nextui-org/react';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Components
import { ArrowDownIcon, ArrowUpIcon, Chip, Text } from '@/components';

export interface IAnalyticsCardItem {
  title: string;
  subtitle: string;
  amount: number;
  percentageChange: number;
  isPositive?: boolean;
}

export const AnalyticsCard = ({
  title,
  subtitle,
  isPositive,
  amount,
  percentageChange,
}: IAnalyticsCardItem) => {
  const ArrowComponent = isPositive ? <ArrowUpIcon /> : <ArrowDownIcon />;

  return (
    <Card key={title} className='w-full gap-4 rounded-md p-0 shadow-none'>
      <CardHeader className='p-4 text-xs font-medium'>
        <Text>{title}</Text>
      </CardHeader>
      <CardBody className='flex flex-row items-center gap-5 overflow-y-hidden p-4'>
        <Text as='span' className='text-[2em] font-extrabold'>
          ${formatNumberWithCommas(amount)}
        </Text>
        <Chip
          startContent={ArrowComponent}
          color={isPositive ? 'success' : 'danger'}
          classNames={{
            base: 'h-5',
            content: 'font-extrabold text-4xs p-0',
          }}
        >
          {percentageChange}%
        </Chip>
      </CardBody>
      <CardFooter className='rounded-none p-4'>
        <Text className='text-2xs font-light text-transparentBlack'>
          {subtitle}
        </Text>
      </CardFooter>
    </Card>
  );
};
