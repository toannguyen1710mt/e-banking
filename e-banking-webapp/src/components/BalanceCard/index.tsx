import { Card } from '@nextui-org/react';

// Utils
import { formatNumberWithCommas } from '@/utils';

// Interfaces
import { TEXT_SIZE, TEXT_VARIANT } from '@/interfaces';

// Components
import { Text, ArrowUpRightIcon, PaymentIcon } from '@/components';

interface BalanceCardProps {
  amount: number;
  title: string;
}

export const BalanceCard = ({ amount, title }: BalanceCardProps) => (
  <Card className='h-[155px] w-full pl-[14px] pr-4 pt-[13px]'>
    <div className='mb-10 flex justify-between text-primary-200'>
      <PaymentIcon />
      <ArrowUpRightIcon />
    </div>

    <div className='flex flex-col gap-3'>
      <Text className='text-2xl font-extrabold text-primary-200' as='span'>
        ${formatNumberWithCommas(amount)}
      </Text>

      <Text size={TEXT_SIZE['2XS']} variant={TEXT_VARIANT.INFO}>
        {title}
      </Text>
    </div>
  </Card>
);
