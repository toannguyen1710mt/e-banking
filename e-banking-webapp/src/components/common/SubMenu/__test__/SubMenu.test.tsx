// Libs
import { render } from '@testing-library/react';

// Component
import { SubMenu } from '@/components';

describe('SubMenu component', () => {
  it('Should render snapshot correctly', () => {
    expect(render(<SubMenu />)).toMatchSnapshot();
  });
});
