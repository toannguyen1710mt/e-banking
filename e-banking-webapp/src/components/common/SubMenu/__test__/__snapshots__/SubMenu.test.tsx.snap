// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubMenu component Should render snapshot correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="border-b border-primary-200 border-opacity-50 pb-4 pl-8 max-[800px]:block hidden"
      >
        <ul
          class="flex gap-8"
        >
          <li
            class=""
          >
            <a
              class="flex w-full items-center justify-start gap-3 border-none text-sm"
              href="/settings/general"
            >
              <svg
                fill="none"
                height="16"
                viewBox="0 0 16 16"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.65063 8.78268L6.18627 8.56808L5.73278 8.21128C5.13365 7.73991 4.69634 7.09351 4.48169 6.36202C4.26703 5.63054 4.28571 4.85033 4.53513 4.12996C4.78454 3.40958 5.25229 2.78485 5.87329 2.34269C6.49429 1.90053 7.23767 1.66292 8 1.66292C8.76233 1.66292 9.50571 1.90053 10.1267 2.34269C10.7477 2.78485 11.2155 3.40958 11.4649 4.12996C11.7143 4.85033 11.733 5.63054 11.5183 6.36202C11.3037 7.09351 10.8663 7.73991 10.2672 8.21128L10.4733 8.47325L10.3504 8.78308C11.4187 9.20708 12.3516 9.91385 13.0489 10.8276C13.7461 11.7412 14.1815 12.8272 14.3086 13.9693C14.3178 14.0566 14.2923 14.144 14.2374 14.2125C14.1858 14.2771 14.1121 14.32 14.0309 14.3333L13.98 14.3332L13.978 14.3333C13.8956 14.3338 13.8159 14.3037 13.7543 14.2489C13.6928 14.1941 13.6537 14.1185 13.6446 14.0366L13.6446 14.036C13.489 12.6514 12.8288 11.3726 11.79 10.4439C10.7512 9.5153 9.4067 9.00194 8.01333 9.00194C6.61997 9.00194 5.27545 9.5153 4.23665 10.4439C3.19786 11.3726 2.53762 12.6514 2.38208 14.036L2.382 14.0368C2.37221 14.1257 2.3275 14.2071 2.2577 14.2631C2.18789 14.3191 2.09871 14.345 2.00978 14.3353C1.92084 14.3255 1.83944 14.2808 1.78347 14.211C1.75575 14.1764 1.73512 14.1367 1.72274 14.0942C1.71037 14.0517 1.70649 14.0071 1.71132 13.9631C1.83775 12.824 2.27089 11.7406 2.9646 10.8283C3.65835 9.91593 4.58668 9.20894 5.65063 8.78268ZM6.33329 7.82766C6.82663 8.15731 7.40665 8.33325 8 8.33325C8.79565 8.33325 9.55871 8.01718 10.1213 7.45457C10.6839 6.89197 11 6.1289 11 5.33325C11 4.73991 10.8241 4.15989 10.4944 3.66654C10.1648 3.1732 9.69623 2.78868 9.14805 2.56162C8.59987 2.33455 7.99667 2.27514 7.41473 2.3909C6.83278 2.50665 6.29823 2.79238 5.87868 3.21193C5.45912 3.63149 5.1734 4.16604 5.05764 4.74798C4.94189 5.32993 5.0013 5.93313 5.22836 6.4813C5.45542 7.02948 5.83994 7.49802 6.33329 7.82766Z"
                  fill="black"
                  stroke="currentColor"
                  stroke-width="0.666667"
                />
              </svg>
              <span>
                General
              </span>
            </a>
          </li>
          <li
            class=""
          >
            <a
              class="flex w-full items-center justify-start gap-3 border-none text-sm"
              href="/settings/account"
            >
              <svg
                fill="none"
                height="18"
                viewBox="0 0 18 18"
                width="18"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M14.9252 9.49504C14.8049 9.35818 14.7386 9.18223 14.7386 9.00004C14.7386 8.81786 14.8049 8.64191 14.9252 8.50504L15.8852 7.42504C15.991 7.30704 16.0566 7.15857 16.0728 7.00092C16.089 6.84328 16.0548 6.68456 15.9752 6.54754L14.4752 3.95254C14.3963 3.81568 14.2763 3.7072 14.1322 3.64256C13.9881 3.57792 13.8273 3.56041 13.6727 3.59254L12.2627 3.87754C12.0833 3.91461 11.8965 3.88474 11.7376 3.79354C11.5787 3.70235 11.4587 3.55616 11.4002 3.38254L10.9427 2.01004C10.8924 1.86108 10.7965 1.73169 10.6686 1.64017C10.5408 1.54866 10.3874 1.49964 10.2302 1.50004H7.23017C7.06662 1.49151 6.90477 1.53674 6.76934 1.62883C6.63391 1.72092 6.53235 1.85481 6.48017 2.01004L6.06017 3.38254C6.00167 3.55616 5.88165 3.70235 5.72275 3.79354C5.56386 3.88474 5.37708 3.91461 5.19767 3.87754L3.75017 3.59254C3.60358 3.57183 3.45414 3.59496 3.32068 3.65902C3.18721 3.72309 3.07569 3.82521 3.00017 3.95254L1.50017 6.54754C1.41854 6.68303 1.38183 6.84086 1.3953 6.99846C1.40877 7.15606 1.47173 7.30537 1.57517 7.42504L2.52767 8.50504C2.64791 8.64191 2.71422 8.81786 2.71422 9.00004C2.71422 9.18223 2.64791 9.35818 2.52767 9.49504L1.57517 10.575C1.47173 10.6947 1.40877 10.844 1.3953 11.0016C1.38183 11.1592 1.41854 11.3171 1.50017 11.4525L3.00017 14.0475C3.07899 14.1844 3.19901 14.2929 3.34311 14.3575C3.48721 14.4222 3.64803 14.4397 3.80267 14.4075L5.21267 14.1225C5.39208 14.0855 5.57886 14.1154 5.73775 14.2065C5.89665 14.2977 6.01667 14.4439 6.07517 14.6175L6.53267 15.99C6.58485 16.1453 6.68641 16.2792 6.82184 16.3713C6.95727 16.4633 7.11912 16.5086 7.28267 16.5H10.2827C10.4399 16.5005 10.5933 16.4514 10.7211 16.3599C10.849 16.2684 10.9449 16.139 10.9952 15.99L11.4527 14.6175C11.5112 14.4439 11.6312 14.2977 11.7901 14.2065C11.949 14.1154 12.1358 14.0855 12.3152 14.1225L13.7252 14.4075C13.8798 14.4397 14.0406 14.4222 14.1847 14.3575C14.3288 14.2929 14.4488 14.1844 14.5277 14.0475L16.0277 11.4525C16.1073 11.3155 16.1415 11.1568 16.1253 10.9992C16.1091 10.8415 16.0435 10.693 15.9377 10.575L14.9252 9.49504ZM13.8077 10.5L14.4077 11.175L13.4477 12.84L12.5627 12.66C12.0225 12.5496 11.4606 12.6414 10.9836 12.9179C10.5066 13.1944 10.1478 13.6364 9.97517 14.16L9.69017 15H7.77017L7.50017 14.145C7.32756 13.6214 6.9687 13.1794 6.49172 12.9029C6.01474 12.6264 5.45283 12.5346 4.91267 12.645L4.02767 12.825L3.05267 11.1675L3.65267 10.4925C4.02163 10.08 4.22562 9.54599 4.22562 8.99254C4.22562 8.43909 4.02163 7.90506 3.65267 7.49254L3.05267 6.81754L4.01267 5.16754L4.89767 5.34754C5.43783 5.45796 5.99974 5.3662 6.47672 5.08969C6.9537 4.81318 7.31256 4.37116 7.48517 3.84754L7.77017 3.00004H9.69017L9.97517 3.85504C10.1478 4.37866 10.5066 4.82068 10.9836 5.09719C11.4606 5.3737 12.0225 5.46546 12.5627 5.35504L13.4477 5.17504L14.4077 6.84004L13.8077 7.51504C13.4428 7.92661 13.2414 8.45756 13.2414 9.00754C13.2414 9.55753 13.4428 10.0885 13.8077 10.5ZM8.73017 6.00004C8.13682 6.00004 7.5568 6.17599 7.06346 6.50564C6.57011 6.83528 6.18559 7.30381 5.95853 7.85199C5.73146 8.40017 5.67205 9.00337 5.78781 9.58531C5.90357 10.1673 6.18929 10.7018 6.60885 11.1214C7.0284 11.5409 7.56295 11.8266 8.1449 11.9424C8.72684 12.0582 9.33004 11.9987 9.87822 11.7717C10.4264 11.5446 10.8949 11.1601 11.2246 10.6668C11.5542 10.1734 11.7302 9.59339 11.7302 9.00004C11.7302 8.20439 11.4141 7.44133 10.8515 6.87872C10.2889 6.31611 9.52582 6.00004 8.73017 6.00004ZM8.73017 10.5C8.43349 10.5 8.14348 10.4121 7.89681 10.2472C7.65014 10.0824 7.45788 9.84816 7.34435 9.57407C7.23081 9.29998 7.20111 8.99838 7.25899 8.70741C7.31687 8.41644 7.45973 8.14916 7.66951 7.93938C7.87928 7.72961 8.14656 7.58674 8.43753 7.52887C8.7285 7.47099 9.0301 7.50069 9.30419 7.61422C9.57828 7.72776 9.81255 7.92001 9.97737 8.16669C10.1422 8.41336 10.2302 8.70337 10.2302 9.00004C10.2302 9.39787 10.0721 9.7794 9.79083 10.0607C9.50952 10.342 9.12799 10.5 8.73017 10.5Z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Account
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="border-b border-primary-200 border-opacity-50 pb-4 pl-8 max-[800px]:block hidden"
    >
      <ul
        class="flex gap-8"
      >
        <li
          class=""
        >
          <a
            class="flex w-full items-center justify-start gap-3 border-none text-sm"
            href="/settings/general"
          >
            <svg
              fill="none"
              height="16"
              viewBox="0 0 16 16"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.65063 8.78268L6.18627 8.56808L5.73278 8.21128C5.13365 7.73991 4.69634 7.09351 4.48169 6.36202C4.26703 5.63054 4.28571 4.85033 4.53513 4.12996C4.78454 3.40958 5.25229 2.78485 5.87329 2.34269C6.49429 1.90053 7.23767 1.66292 8 1.66292C8.76233 1.66292 9.50571 1.90053 10.1267 2.34269C10.7477 2.78485 11.2155 3.40958 11.4649 4.12996C11.7143 4.85033 11.733 5.63054 11.5183 6.36202C11.3037 7.09351 10.8663 7.73991 10.2672 8.21128L10.4733 8.47325L10.3504 8.78308C11.4187 9.20708 12.3516 9.91385 13.0489 10.8276C13.7461 11.7412 14.1815 12.8272 14.3086 13.9693C14.3178 14.0566 14.2923 14.144 14.2374 14.2125C14.1858 14.2771 14.1121 14.32 14.0309 14.3333L13.98 14.3332L13.978 14.3333C13.8956 14.3338 13.8159 14.3037 13.7543 14.2489C13.6928 14.1941 13.6537 14.1185 13.6446 14.0366L13.6446 14.036C13.489 12.6514 12.8288 11.3726 11.79 10.4439C10.7512 9.5153 9.4067 9.00194 8.01333 9.00194C6.61997 9.00194 5.27545 9.5153 4.23665 10.4439C3.19786 11.3726 2.53762 12.6514 2.38208 14.036L2.382 14.0368C2.37221 14.1257 2.3275 14.2071 2.2577 14.2631C2.18789 14.3191 2.09871 14.345 2.00978 14.3353C1.92084 14.3255 1.83944 14.2808 1.78347 14.211C1.75575 14.1764 1.73512 14.1367 1.72274 14.0942C1.71037 14.0517 1.70649 14.0071 1.71132 13.9631C1.83775 12.824 2.27089 11.7406 2.9646 10.8283C3.65835 9.91593 4.58668 9.20894 5.65063 8.78268ZM6.33329 7.82766C6.82663 8.15731 7.40665 8.33325 8 8.33325C8.79565 8.33325 9.55871 8.01718 10.1213 7.45457C10.6839 6.89197 11 6.1289 11 5.33325C11 4.73991 10.8241 4.15989 10.4944 3.66654C10.1648 3.1732 9.69623 2.78868 9.14805 2.56162C8.59987 2.33455 7.99667 2.27514 7.41473 2.3909C6.83278 2.50665 6.29823 2.79238 5.87868 3.21193C5.45912 3.63149 5.1734 4.16604 5.05764 4.74798C4.94189 5.32993 5.0013 5.93313 5.22836 6.4813C5.45542 7.02948 5.83994 7.49802 6.33329 7.82766Z"
                fill="black"
                stroke="currentColor"
                stroke-width="0.666667"
              />
            </svg>
            <span>
              General
            </span>
          </a>
        </li>
        <li
          class=""
        >
          <a
            class="flex w-full items-center justify-start gap-3 border-none text-sm"
            href="/settings/account"
          >
            <svg
              fill="none"
              height="18"
              viewBox="0 0 18 18"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M14.9252 9.49504C14.8049 9.35818 14.7386 9.18223 14.7386 9.00004C14.7386 8.81786 14.8049 8.64191 14.9252 8.50504L15.8852 7.42504C15.991 7.30704 16.0566 7.15857 16.0728 7.00092C16.089 6.84328 16.0548 6.68456 15.9752 6.54754L14.4752 3.95254C14.3963 3.81568 14.2763 3.7072 14.1322 3.64256C13.9881 3.57792 13.8273 3.56041 13.6727 3.59254L12.2627 3.87754C12.0833 3.91461 11.8965 3.88474 11.7376 3.79354C11.5787 3.70235 11.4587 3.55616 11.4002 3.38254L10.9427 2.01004C10.8924 1.86108 10.7965 1.73169 10.6686 1.64017C10.5408 1.54866 10.3874 1.49964 10.2302 1.50004H7.23017C7.06662 1.49151 6.90477 1.53674 6.76934 1.62883C6.63391 1.72092 6.53235 1.85481 6.48017 2.01004L6.06017 3.38254C6.00167 3.55616 5.88165 3.70235 5.72275 3.79354C5.56386 3.88474 5.37708 3.91461 5.19767 3.87754L3.75017 3.59254C3.60358 3.57183 3.45414 3.59496 3.32068 3.65902C3.18721 3.72309 3.07569 3.82521 3.00017 3.95254L1.50017 6.54754C1.41854 6.68303 1.38183 6.84086 1.3953 6.99846C1.40877 7.15606 1.47173 7.30537 1.57517 7.42504L2.52767 8.50504C2.64791 8.64191 2.71422 8.81786 2.71422 9.00004C2.71422 9.18223 2.64791 9.35818 2.52767 9.49504L1.57517 10.575C1.47173 10.6947 1.40877 10.844 1.3953 11.0016C1.38183 11.1592 1.41854 11.3171 1.50017 11.4525L3.00017 14.0475C3.07899 14.1844 3.19901 14.2929 3.34311 14.3575C3.48721 14.4222 3.64803 14.4397 3.80267 14.4075L5.21267 14.1225C5.39208 14.0855 5.57886 14.1154 5.73775 14.2065C5.89665 14.2977 6.01667 14.4439 6.07517 14.6175L6.53267 15.99C6.58485 16.1453 6.68641 16.2792 6.82184 16.3713C6.95727 16.4633 7.11912 16.5086 7.28267 16.5H10.2827C10.4399 16.5005 10.5933 16.4514 10.7211 16.3599C10.849 16.2684 10.9449 16.139 10.9952 15.99L11.4527 14.6175C11.5112 14.4439 11.6312 14.2977 11.7901 14.2065C11.949 14.1154 12.1358 14.0855 12.3152 14.1225L13.7252 14.4075C13.8798 14.4397 14.0406 14.4222 14.1847 14.3575C14.3288 14.2929 14.4488 14.1844 14.5277 14.0475L16.0277 11.4525C16.1073 11.3155 16.1415 11.1568 16.1253 10.9992C16.1091 10.8415 16.0435 10.693 15.9377 10.575L14.9252 9.49504ZM13.8077 10.5L14.4077 11.175L13.4477 12.84L12.5627 12.66C12.0225 12.5496 11.4606 12.6414 10.9836 12.9179C10.5066 13.1944 10.1478 13.6364 9.97517 14.16L9.69017 15H7.77017L7.50017 14.145C7.32756 13.6214 6.9687 13.1794 6.49172 12.9029C6.01474 12.6264 5.45283 12.5346 4.91267 12.645L4.02767 12.825L3.05267 11.1675L3.65267 10.4925C4.02163 10.08 4.22562 9.54599 4.22562 8.99254C4.22562 8.43909 4.02163 7.90506 3.65267 7.49254L3.05267 6.81754L4.01267 5.16754L4.89767 5.34754C5.43783 5.45796 5.99974 5.3662 6.47672 5.08969C6.9537 4.81318 7.31256 4.37116 7.48517 3.84754L7.77017 3.00004H9.69017L9.97517 3.85504C10.1478 4.37866 10.5066 4.82068 10.9836 5.09719C11.4606 5.3737 12.0225 5.46546 12.5627 5.35504L13.4477 5.17504L14.4077 6.84004L13.8077 7.51504C13.4428 7.92661 13.2414 8.45756 13.2414 9.00754C13.2414 9.55753 13.4428 10.0885 13.8077 10.5ZM8.73017 6.00004C8.13682 6.00004 7.5568 6.17599 7.06346 6.50564C6.57011 6.83528 6.18559 7.30381 5.95853 7.85199C5.73146 8.40017 5.67205 9.00337 5.78781 9.58531C5.90357 10.1673 6.18929 10.7018 6.60885 11.1214C7.0284 11.5409 7.56295 11.8266 8.1449 11.9424C8.72684 12.0582 9.33004 11.9987 9.87822 11.7717C10.4264 11.5446 10.8949 11.1601 11.2246 10.6668C11.5542 10.1734 11.7302 9.59339 11.7302 9.00004C11.7302 8.20439 11.4141 7.44133 10.8515 6.87872C10.2889 6.31611 9.52582 6.00004 8.73017 6.00004ZM8.73017 10.5C8.43349 10.5 8.14348 10.4121 7.89681 10.2472C7.65014 10.0824 7.45788 9.84816 7.34435 9.57407C7.23081 9.29998 7.20111 8.99838 7.25899 8.70741C7.31687 8.41644 7.45973 8.14916 7.66951 7.93938C7.87928 7.72961 8.14656 7.58674 8.43753 7.52887C8.7285 7.47099 9.0301 7.50069 9.30419 7.61422C9.57828 7.72776 9.81255 7.92001 9.97737 8.16669C10.1422 8.41336 10.2302 8.70337 10.2302 9.00004C10.2302 9.39787 10.0721 9.7794 9.79083 10.0607C9.50952 10.342 9.12799 10.5 8.73017 10.5Z"
                fill="currentColor"
              />
            </svg>
            <span>
              Account
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
